$OpenBSD: patch-test_torrent_tracker_controller_test_cc,v 1.1 2012/07/11 08:33:39 dcoppa Exp $

Fix a bug where manual tracker update for a preferred tracker that
fails would end up rerequesting instantly (test)
(upstream git commit 78f56ee74cecd8e82d39baaea10395301fbec4b8)

--- test/torrent/tracker_controller_test.cc.orig	Tue Apr 10 14:46:33 2012
+++ test/torrent/tracker_controller_test.cc	Tue Jul 10 16:33:20 2012
@@ -237,6 +237,19 @@ tracker_controller_test::test_send_update_normal() {
 }
 
 void
+tracker_controller_test::test_send_update_failure() {
+  torrent::cachedTime = rak::timer::from_seconds(1 << 20);
+  TEST_SINGLE_BEGIN();
+
+  tracker_controller.send_update_event();
+
+  TEST_SINGLE_FAILURE_TIMEOUT(5);
+  TEST_SINGLE_FAILURE_TIMEOUT(10);
+
+  TEST_SINGLE_END(0, 2);
+}
+
+void
 tracker_controller_test::test_send_task_timeout() {
   TEST_SINGLE_BEGIN();
   TEST_SEND_SINGLE_BEGIN(update);
